server/src/main/java/org/elasticsearch/ingest/IngestDocument.java

```diff
@@ -104,6 +104,7 @@ public IngestDocument(IngestDocument other) {
             new IngestCtxMap(deepCopyMap(ensureNoSelfReferences(other.ctxMap.getSource())), other.ctxMap.getMetadata().clone()),
             deepCopyMap(other.ingestMetadata)
         );
+        this.executedPipelines.addAll(other.executedPipelines);
```

Comment:
I think a mea culpa comment about why we're doing this would be a good addition.